pkg/sshd/circuit.go

```diff
@@ -0,0 +1,61 @@
+package sshd
+
+import (
+	"fmt"
+	"sync/atomic"
+)
+
+// CircuitState represents the state of a Circuit
+type CircuitState uint32
```

Comment:
Yup, good question.

A `bool`/`sync.RWMutex` or channel-based implementation would work just as well here, and both would fit fine into this abstraction. I went the `sync/atomic` route mainly because there's no need to protect a _critical section_ of logic - just the mutation of a single boolean - and `sync/atomic` can do concurrency-safe bit-twiddling well.

Basically, I originally checked this code in thinking it would be simplest and _not_ because I had any performance or specific `sync/atomic` benefits in mind. Now that I've gotten a few questions on it, I wonder if it'd be best to refactor it using a mutex, which seems to be more familiar. Thoughts?

_Here's some more on `sync/atomic` if you care:_

Atomic integer and pointer manipulation usually does lend itself to lower level systems problems like databases, etc..., but the core concept still applies here - atomically manipulating bits. We're just approximating that single bit as a `uint32`.

Also, on many systems, the `sync/atomic` approach is more performant in a few different regards, but that's getting into scheduler and OS level implementation details, which is where I get fuzzy anyway.